File: official/docs/php/end_shipper/BuyShipmentWithEndShipper.php
<?php

\EasyPost\EasyPost::setApiKey($_ENV['EASYPOST_API_KEY']);

$shipment = \EasyPost\Shipment::retrieve('shp_...');

$shipment->buy([
    'rate'      => 'rate_...',
    'insurance' => null,
    'with_carbon_offset' => false,
    'end_shipper_id' => 'es_...'
]);

echo $shipment;

File: official/docs/java/endShipper/BuyShipmentWithEndShipper.java
import com.easypost.EasyPost;
import java.util.HashMap;

public class BuyShipmentWithEndShipper {
    public static void main(String[] args) {
        EasyPost.apiKey = System.getenv("EASYPOST_API_KEY");

        Shipment shipment = Shipment.retrieve("shp_...");

        HashMap<String, Object> buyMap = new HashMap<String, Object>();
        buyMap.put("rate", shipment.lowestRate());
        buyMap.put("end_shipper_id", "es_...");

        shipment.buy(buyMap);

        System.out.println(shipment);
    }
}
